train/tcn_train.py

```diff
@@ -0,0 +1,564 @@
+    plt.xlabel('Epoch')
+    plt.ylabel('Accuracy')
+    plt.savefig(f'{parameters.results_path}{train_name}_accuracy_{datetime.date(datetime.now())}_run_{run}.png')
+    plt.close()
+
+    # plot losses
+    # plt.figure(3+run)
+    plt.figure()
+    plt.plot(train_label_losses, 'or', label='train label loss')
+    plt.plot(train_plant_losses, 'ob', label='train plant loss')
+    plt.plot(test_losses, 'og', label='test label loss')
+    plt.title(f'{parameters.experiment} Train and Test Losses')
+    plt.legend(title='Parameters')
+    plt.xlabel('Epoch')
+    plt.ylabel('Loss')
+    plt.savefig(f'{parameters.results_path}{train_name}_loss_{datetime.date(datetime.now())}_run_{run}.png')
+    plt.close()
+
+    conf_matrix, class_report = confusion_matrix(test_config, run)
+
+    return train_label_losses, train_plant_losses, train_accuracy_prog, \
+                test_accuracy, test_losses, conf_matrix, class_report
+
+
+def restore_checkpoint(test_config: TestConfig):
+    checkpoint = torch.load(f'checkpoints/{test_config.checkpoint_name}')
+    test_config.feat_ext.load_state_dict(checkpoint['feat_ext_state_dict'])
+    test_config.label_cls.load_state_dict(checkpoint['label_cls_state_dict'])
+    test_config.plant_cls.load_state_dict(checkpoint['plant_cls_state_dict'])
+    test_config.best_loss = checkpoint['loss']
+
+    print(f"Restoring model to one with loss - {test_config.best_loss}")
+
+    test_config.feat_ext = test_config.feat_ext.to(test_config.device)
+    test_config.label_cls = test_config.label_cls.to(test_config.device)
+    test_config.plant_cls = test_config.plant_cls.to(test_config.device)
+
+
+def main():
+    checkpoint_name = get_checkpoint_name(parameters.experiment, parameters.excluded_modalities)
+
+    device = 'cuda' if torch.cuda.is_available() else 'cpu'
+
+    used_modalities = parameters.used_modalities
+    inception_embeddings_path = parameters.inception_embeddings_path
+
+    start_date = experiments_info[parameters.experiment].start_date
+    end_date = experiments_info[parameters.experiment].end_date
+    if parameters.num_days is not None:
+        end_date = experiments_info[parameters.experiment].start_date + timedelta(days=parameters.num_days - 1)
+
+    # creating plants file dictionary # TODO: could be uploaded
+    plant_files_dict = defaultdict(dict)
+    mod_receptive_field = {}
+    num_plants = len(labels[parameters.experiment])
+    used_plants = [plant for plant in range(num_plants)]
+    root_dir = '/home/pnn/inception_features_exp0'  # TODO change to generic root dir definition
+
+    global y_tot, y_hat_tot
+    y_tot = []
+    y_hat_tot = []
+
+    for mod in parameters.used_modalities:
+        for plant in used_plants:
+            plant_name = f'plant_{plant}_'  # example plant
+            files = sorted(glob.glob(f'{root_dir}/*{mod}*{plant_name}*'))
+            if plant is 0:
+                mod_receptive_field[mod] = len(files)
```

Comment:
- [ ] since you want to add the receptive field if it does not exist, it'll be better to make the condition reflect it.
```suggestion
            if mod not in mod_receptive_field:
                mod_receptive_field[mod] = len(files)
```